credential/base/src/main/java/org/wildfly/security/credential/SSHLocationCredential.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.credential;
+
+import java.io.File;
+import java.util.Objects;
+
+/**
+ * A credential holding the location, key identity and passphrase (instance of {@code Credential}) of a Private key in an external
+ * file and/or the file name of containing the known hosts
+ *
+ * @author <a href="mailto:aabdelsa@redhat.com">Ashley Abdel-Sayed</a>
+ */
+public class SSHLocationCredential implements Credential {
+
+    private final File homeDirectory;
+    private final File sshDirectory;
+    private final String privateKeyIdentity;
+    private final Credential passphrase;
+    private final String knownHostsFile;
+
+    /**
+     * @param homeDirectory the home directory
+     * @param sshDirectory the ssh directory
+     * @param privateKeyIdentity the name of the file containing the private key
+     * @param passphrase the passphrase used to encrypt the private key
+     * @param knownHostsFile the name of the file containing the known hosts
+     */
+    public SSHLocationCredential(File homeDirectory, File sshDirectory, String privateKeyIdentity, Credential passphrase, String knownHostsFile) {
```

Comment:
It would be good to use the Java `Builder` pattern for this. See https://dzone.com/articles/design-patterns-the-builder-pattern.

Reply:
Fixed, thanks!